openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -218,8 +218,10 @@ def get_all_drafts(learning_package_id: int, /) -> QuerySet[Draft]:
 
 def get_entities_with_unpublished_changes(learning_package_id: int, /) -> QuerySet[PublishableEntity]:
     return PublishableEntity.objects \
-                            .filter(learning_package_id=learning_package_id) \
-                            .exclude(draft__version=F('published__version'))
+                            .filter(
+                                learning_package_id=learning_package_id,
+                                draft__version__isnull=False,
+                            ).exclude(draft__version=F('published__version'))
```

Comment:
Please add a docstring explaining this behavior (that it omits the entities that were soft-deleted), or make it a keyword argument. Otherwise, I'm fine with this.

Question that this brought to mind (doesn't have to be part of this PR): Does [publish_all_drafts](https://github.com/openedx/openedx-learning/blob/835c9a1894fb7de5749d3ef9b5e495e889ec3384/openedx_learning/apps/authoring/publishing/api.py#L248-L251) have a bug where it's re-publishing soft-deletes over and over again because NULL != NULL in SQL?

Reply:
> Please add a docstring explaining this behavior (that it omits the entities that were soft-deleted), or make it a keyword argument. Otherwise, I'm fine with this.

Done.

> Question that this brought to mind (doesn't have to be part of this PR): Does [publish_all_drafts](https://github.com/openedx/openedx-learning/blob/835c9a1894fb7de5749d3ef9b5e495e889ec3384/openedx_learning/apps/authoring/publishing/api.py#L248-L251) have a bug where it's re-publishing soft-deletes over and over again because NULL != NULL in SQL?

Yes, it does re-publish it over and over again. We probably need to use a specific value in version field like `-1` for soft-deletes.

Reply:
I think we can stop the duplicate publishing by excluding Draft records that are null, without resorting to special version field values. I'll make a PR for it.